bundles/org.openhab.binding.wemo/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -162,6 +162,32 @@
 		</config-description>
 	</thing-type>
 
+     <thing-type id="dimmer">
+        <label>WeMo DimmerSwitch</label>
+        <description>This is a WeMo DimmerSwitch</description>
+
+        <channels>
+            <channel id="state" typeId="state"/>

```

Comment:
A dimmer must not have such a switch channel - the `brightness` channel service this purpose already.

Reply:
This device has deinitely a separat ON/OFF funktion. You can change it's brightness without switching it. Weired, but that's Belkin ...

Reply:
That's the same for hue and others. But you have to map this to the openHAB model: 0% simply means the device is OFF, so if you receive 0% as a brightness command, make sure you switch the dimmer off. And for any other value turn it on. But don't have such a `state` channel on the openHAB model side, please.

Reply:
I will heave to check with the device directly. It physically acts different.

Reply:
It's the same with Philips hue and others, see e.g. https://github.com/openhab/openhab2-addons/blob/master/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueLightHandler.java#L257-L267.

Reply:
I just checked with the device physically and the original Wemo app. Sorry to say, that the WeMo DommerSwitch (the name is on purpose) acts different and does not fit into this. You cannot set the brightness to 0 (a have to catch this in the handler) and changing the brightness value does not switch the device on. openHAB treats changing the brigthness as switching the device on. So we run into a situation where openHAB is not reflecting the physical state of the device.

Reply:
That's why I said that the behavior needs to be mapped to the "openHAB model". It is the task of the binding to switch the dimmer on, if the openHAB user wishes a brightness of 50%. We do this in ALL bindings this way: hue, LIFX, etc. As a result, the user has the same experience, independent of the product - that's the power of the abstraction layer.

Reply:
But what if the users wishe just to change the brithness wuthout switching the device on ? Think of changing a preset value.

Reply:
Then you are talking about a brightness preset value and not about the brightness channel anymore. The semantic of the brightness channel simply is that the dimmer changes to the given brightness.
For hue and others, the brightness channel has a configuration parameter like "powerOnBrightness", which defines what brightness should be used when receiving an "ON" command.

Reply:
I guess we are running into a philosophical discussion now. This device does not completely fit into openHABs semantic. 
I agree, the brightness channel simply changes the brightness of the dimmer. But this device can have its brightness changed even it is switched off. There are still some small LEDs on the device indicating the brightness level when it is switched off. So letting the binding act like HUE, LFX or whatever, we can run into a situation where the binding state could differ from the physical state.

Reply:
It's not philosophical, it is very pragmatic: The behaviour of a brightness channel of a dimmer must be as described, so that it is the same across different bindings. There is no problem in writing the binding in a way that this works with the Wemo hardware. I don't think it is necessary to have another channel representing the small LEDs - I don't see much value in that.

Reply:
Foget about these LEDs, the issue would that one could change the brightness through WeMo App or physically and this would lead to a wrong on/off state in openHAB.
The WeMo Dimmer Switch is somehow working as two devices in one box, a Switch and a Dimmer.

Reply:
> the issue would that one could change the brightness through WeMo App or physically and this would lead to a wrong on/off state in openHAB.

Why? Sorry, I still do not understand the issue. If the switch is on, the brightness will be the same as in openHAB (and the state of the brightness channel can be updated if changed externally). If it is off, the state of the brightness channel in openHAB is OFF and it stays OFF, if the preset changes externally. So all in sync, isn't it?

Reply:
No it isn't. If the brightness value changes while the device is switched off, openhab sets brightness to on, but the device physically is still switched off.

Reply:
If this is what is happening, I would say it is a bug in the binding implementation. As I said: It is the duty of the binding to map the behavior to the semantic model of a brightness channel in openHAB. So if the wemo-brightness value changes, while the dimmer is switched off, there must be no change at all on the `brightness` channel in openHAB, since the dimmer is still OFF (=0% for openHAB). And as you say that you are not interested in the state of the small LEDs, there is no need for openHAB to notice the change at all. If an "ON" command is sent, the dimmer will set the (real) brightness to the right value by itself, openHAB does not have to know it. Or am I missing something?

Reply:
Hm, still not convinced but I will give it some more thoughts to find the best fit for both sides.
I guess I have to find a way to store a changed brightness value and map an ON command to set this value as openHABs brightness value and simply send the binarystate ON to the device.
Let me see ...

Reply:
My hope would have been that if you send an ON command to the device that you then immediately get the current brightness back from it as a state update. Isn't this the case?

Reply:
Unfortunately not, UPnP events are not working for the brightness value, so an update would arrive with the next polling cycle.

Reply:
Ok, so that's the same as with Philips hue actually. Imho, it won't happen too often that the user changes the dimmer preset by some other means and then wonders that the value takes a few seconds to show in openHAB when turning it on the next time. So I'd think we can easily live with that behaviour.

Reply:
State channel removed an brightness synchronized